WolvenKit/Views/Documents/RDTDataView.xaml.cs

```diff
@@ -42,87 +31,26 @@ public RDTDataView()
-                //       viewmodel => viewmodel.SelectedChunk.PropertyGridItems,
-                //       view => view.PropertyGrid.Items)
-                //   .DisposeWith(disposables);
-
-                //this.OneWayBind(ViewModel,
-                //       viewmodel => viewmodel.SelectedChunk,
-                //       view => view.CustomPG.DataContext)
-                //   .DisposeWith(disposables);
-
-                //this.ViewModel.WhenAnyValue(x => x.SelectedChunk)
-                //    .Where(x => x != null)
-                //    .Select(x => new ObservableCollection<PropertyGridItem>(x.Properties
-                //        .Where(x => x != null)
-                //        .Select(x => new PropertyGridItem()
-                //        {
-                //            PropertyName = x.Name,
-                //            Editor = PropertyGridEditors.GetPropertyEditor(x.PropertyType)
-                //        }
-                //    ))).BindTo(PropertyGrid, x => x.Items);
-
-                //this.BindCommand(ViewModel, vm => vm.ExportChunkCommand, v => v.ExportChunkCommand)
-                //    .DisposeWith(disposables);
-
-                //this.OneWayBind(ViewModel,
-                //       viewmodel => viewmodel.SelectedChunk.Name,
-                //       view => view.PropertyGrid.SelectedPropertyItem.DisplayName)
-                //   .DisposeWith(disposables);
-
+                RedTreeView.PropertyChanged += RedTreeView_OnPropertyChanged;
             });
-            
-
-
-            //PropertyGrid.CustomEditorCollection = CustomEditorCollection;
-            //MainTreeGrid.RequestTreeItems += TreeGrid_RequestTreeItems;
         }
 
-        //public ICommand AddItemToArrayCommand { get; private set; }
-        //public ICommand ExportChunkCommand { get; private set; }
-
-        //private void HandleTemplateView_OnGoToChunkRequested(object sender, GoToChunkRequestedEventArgs e)
-        //{
-        //    var target = e.Export;
-
-        //    if (ViewModel == null || target == null)
-        //    {
-        //        return;
-        //    }
-
-        //    var chunk = ViewModel.Chunks.FirstOrDefault(x => x.Name.Equals(target.REDName));
-        //    chunk.IsSelected = true;
-        //    ViewModel.SelectedChunk = chunk;
-        //}
+        /// <summary>
+        /// We need to sync our view model's selection with the tree. For some reason,
+        /// that doesn't happen automatically anymore.
+        /// </summary>
+        private void RedTreeView_OnPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (sender is not RedTreeView tree || e.PropertyName is not nameof(RedTreeView.SelectedItems) ||
+                ViewModel is null)
+            {
+                return;
+            }
 
+            ViewModel.SelectedChunks ??= [];
+            ViewModel.SelectedChunks.Clear();
+            ViewModel.SelectedChunks.AddRange(tree.SelectedItems.ToList());
```

Comment:
This is complete and utter bullshit. I have to shove them up the list in this event because the property doesn't update – but if I use SetProperty, then the two-way binding fires and nulls the tree's selection??? What the actual fuck! @rfuzzo help